### PR TITLE
TASK-54694 : fix loading winners

### PR DESCRIPTION
**Pull request by GouadriaHanen**

### CODE REVIEW

File: services/src/main/java/org/exoplatform/addons/gamification/rest/AnnouncementRest.java [hunk 1]
```diff
@@ -88,11 +100,24 @@ public Response getAllAnnouncementByChallenge(@ApiParam(value = "id of the chall
     if (limit <= 0) {
       return Response.status(Response.Status.BAD_REQUEST).entity("Limit must be positive").build();
     }
+    EntityTag eTag = null ;
+    List<AnnouncementRestEntity> announcementsRestEntities = new ArrayList<>();
+    eTag = new EntityTag(String.valueOf(announcementsRestEntities.hashCode()));
```
Comment (ahamdi) [inline services/src/main/java/org/exoplatform/addons/gamification/rest/AnnouncementRest.java]: unneeded, you have to calculate Etag once you have loaded the entities to cache the response.

File: services/src/main/java/org/exoplatform/addons/gamification/rest/AnnouncementRest.java [hunk 2]
```diff
@@ -24,10 +28,16 @@
...
   private static final Log    LOG = ExoLogger.getLogger(AnnouncementRest.class);
 
+  private final CacheControl cacheControl;
+
+
   private AnnouncementService announcementService;
 
   public AnnouncementRest(AnnouncementService announcementService) {
     this.announcementService = announcementService;
+    this.cacheControl = new CacheControl();
```
Comment (ahamdi) [inline services/src/main/java/org/exoplatform/addons/gamification/rest/AnnouncementRest.java]: Are you using this ? why did you add it ?

**Review by ahamdi — CHANGES_REQUESTED** [review]

File: services/src/main/java/org/exoplatform/addons/gamification/rest/AnnouncementRest.java (hunk 1 above)
Comment (GouadriaHanen) [reply services/src/main/java/org/exoplatform/addons/gamification/rest/AnnouncementRest.java]: done!

File: services/src/main/java/org/exoplatform/addons/gamification/rest/AnnouncementRest.java (hunk 2 above)
Comment (GouadriaHanen) [reply services/src/main/java/org/exoplatform/addons/gamification/rest/AnnouncementRest.java]: cleaned up!

**Review by ahamdi — APPROVED** [review]
